src/main/java/nextstep/subway/entity/group/SectionGroup.java

```diff
@@ -164,12 +174,59 @@ private void validateSizeCanBeDeleted() {
         }
     }
 
-    private void validateDeleteStationIsEndStation(long deleteStationId) {
-        if (!getEndDownStation().isEqualsDownStation(deleteStationId)) {
-            throw new IllegalArgumentException("하행 종점역이 아니면 삭제할 수 없습니다.");
+    private void validateExistCanBeDeleted(long deleteStationId) {
+
+        boolean noneExistStationInLine = sections.stream()
+            .noneMatch(section
+                -> section.isEqualsUpStation(deleteStationId)
+                || section.isEqualsDownStation(deleteStationId)
+            );
```

Comment:
구간의 상행역 또는 하행역에 포함되는지 판단하는 중복 코드가 많은 것 같아요
Section 객체에 책임을 위임하고 메시지를 보내면 어떨까요?
```suggestion
            .noneMatch(section -> section.contains(deleteStationid));
```

Reply:
너무 좋은 의견이네요!